Makefile

```diff
@@ -65,37 +68,76 @@ endif
+	apt-get update && apt-get install bats && apt-get install gettext-base -y
+	# Download and install Vault
+	curl -LO https://releases.hashicorp.com/vault/${VAULT_VERSION}/vault_${VAULT_VERSION}_linux_amd64.zip && unzip vault_${VAULT_VERSION}_linux_amd64.zip && chmod +x vault && mv vault /usr/local/bin/
+ifndef TEST_WINDOWS
 	curl -LO https://storage.googleapis.com/kubernetes-release/release/v${KUBERNETES_VERSION}/bin/linux/amd64/kubectl && chmod +x ./kubectl && mv kubectl /usr/local/bin/
+	make setup-kind
+endif
+	docker pull $(IMAGE_TAG) || make e2e-container
+
+.PHONY: setup-kind
+setup-kind:
 	# Download and install kind
 	curl -L https://github.com/kubernetes-sigs/kind/releases/download/v${KIND_VERSION}/kind-linux-amd64 --output kind && chmod +x kind && mv kind /usr/local/bin/
-	# Download and install Helm
-	curl https://raw.githubusercontent.com/helm/helm/master/scripts/get | bash -s -- --version ${HELM_VERSION}
-	# Download and install Vault
-	curl -LO https://releases.hashicorp.com/vault/${VAULT_VERSION}/vault_${VAULT_VERSION}_linux_amd64.zip && unzip vault_${VAULT_VERSION}_linux_amd64.zip && chmod +x vault && mv vault /usr/local/bin/
 	# Create kind cluster
 	kind create cluster --config kind-config.yaml --image kindest/node:v${KUBERNETES_VERSION}
-	# Build image
-	REGISTRY_NAME="e2e" IMAGE_VERSION=${E2E_IMAGE_VERSION} make image
-	# Load image into kind cluster
-	kind load docker-image --name kind e2e/secrets-store-csi:${E2E_IMAGE_VERSION}
-	# Set up tiller
-	kubectl --namespace kube-system --output yaml create serviceaccount tiller --dry-run | kubectl --kubeconfig $$(kind get kubeconfig-path)  apply -f -
-	kubectl create --output yaml clusterrolebinding tiller-cluster-rule --clusterrole=cluster-admin --serviceaccount=kube-system:tiller --dry-run | kubectl --kubeconfig $$(kind get kubeconfig-path) apply -f -
-	helm init --service-account tiller --upgrade --wait --kubeconfig $$(kind get kubeconfig-path)
+
+.PHONY: e2e-container
+e2e-container:
+ifdef TEST_WINDOWS
+		az acr login --name $(REGISTRY_NAME)
+		make build build-windows
+		az acr build --registry $(REGISTRY_NAME) -t $(IMAGE_TAG)-linux-amd64 -f Dockerfile --platform linux .
+		az acr build --registry $(REGISTRY_NAME) -t $(IMAGE_TAG)-windows-1809-amd64 -f windows.Dockerfile --platform windows .
+		docker manifest create $(IMAGE_TAG) $(IMAGE_TAG)-linux-amd64 $(IMAGE_TAG)-windows-1809-amd64
+		docker manifest inspect $(IMAGE_TAG)
+		docker manifest push --purge $(IMAGE_TAG)
+else
+		REGISTRY="e2e" make image
+		kind load docker-image --name kind e2e/secrets-store-csi:${IMAGE_VERSION}
+endif
+
+.PHONY: install-helm
+install-helm:
+	curl https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 | bash
+
+.PHONY: e2e-teardown
+e2e-teardown:
+	helm delete csi-secrets-store --namespace default
+
+.PHONY: install-driver
+install-driver:
+ifdef TEST_WINDOWS
+		helm install csi-secrets-store charts/secrets-store-csi-driver --namespace default --wait --timeout=15m -v=5 --debug \
```

Comment:
maybe use the same helm install cmd but set diff variables in the if/else?

Reply:
The variables for linux and windows are different which is why the helm install cmd is different in if/else